src/operators.jl

```diff
@@ -81,16 +82,21 @@ end
 
 function (w::OpWrapper{F,T})(_a::Ptr{Cvoid}, _b::Ptr{Cvoid}, _len::Ptr{Cint}, t::Ptr{MPI_Datatype}) where {F,T}
     len = unsafe_load(_len)
-    @assert isconcretetype(T)
-    a = Ptr{T}(_a)
-    b = Ptr{T}(_b)
-    for i = 1:len
-        unsafe_store!(b, w.f(unsafe_load(a,i), unsafe_load(b,i)), i)
+    if !isconcretetype(T)
+        T = to_type(Datatype(unsafe_load(t))) # Ptr might actually point to a Julia object so we could unsafe_pointer_to_objref?
     end
+    function copy(::Type{T}) where T
+        @assert isconcretetype(T)
+        a = Ptr{T}(_a)
+        b = Ptr{T}(_b)
+        for i = 1:len
+            unsafe_store!(b, w.f(unsafe_load(a,i), unsafe_load(b,i)), i)
+        end
+    end
+    copy(T)
```

Comment:
With
```suggestion
    concrete_T = isconcretetype(T) ? T : to_type(Datatype(unsafe_load(t))) # Ptr might actually point to a Julia object so we could unsafe_pointer_to_objref?
    function copy(::concrete_Tpe{concrete_T}) where concrete_T
        @assert isconcretetype(concrete_T)
        a = Ptr{concrete_T}(_a)
        b = Ptr{concrete_T}(_b)
        for i = 1:len
            unsafe_store!(b, w.f(unsafe_load(a,i), unsafe_load(b,i)), i)
        end
    end
    copy(concrete_T)
```
tests pass for me on aarch64-darwin

Reply:
Also, tests pass for me on this platform including https://github.com/JuliaParallel/MPI.jl/blob/aac9688e6961bc7e3aeeba7600f5e7d0b10596a3/test/test_reduce.jl#L120-L131

Reply:
Yeah!